Sources/TuistKit/Services/Inspect/InspectBuildService.swift

```diff
@@ -0,0 +1,191 @@
+            // analytics to the server.
+            try backgroundProcessRunner.runInBackground(
+                [executablePath, "inspect", "build"],
+                environment: environment
+            )
+            return
+        }
+        let projectPath = try await projectPath(path)
+        let buildLogsPath = try derivedDataLocator.locate(for: projectPath)
+            .appending(components: "Logs", "Build")
+        let logManifestPlistPath = buildLogsPath.appending(component: "LogStoreManifest.plist")
+
+        guard try await fileSystem.exists(logManifestPlistPath)
+        else {
+            throw InspectBuildServiceError.noBuildLogFound(buildLogsPath: buildLogsPath, projectPath: projectPath)
+        }
+        let xcactivityLog = try await latestXCActivityLog(
+            logManifestPlistPath: logManifestPlistPath,
+            buildLogsPath: buildLogsPath,
+            projectPath: projectPath,
+            referenceDate: referenceDate
+        )
+        try await createBuild(
+            for: xcactivityLog,
+            projectPath: projectPath
+        )
+    }
+
+    private func createBuild(
+        for xcactivityLog: XCActivityLog,
+        projectPath: AbsolutePath
+    ) async throws {
+        let config = try await configLoader
+            .loadConfig(path: projectPath)
+        guard let fullHandle = config.fullHandle else { throw InspectBuildServiceError.missingFullHandle }
+        try await createBuildService.createBuild(
+            fullHandle: fullHandle,
+            serverURL: config.url,
+            id: xcactivityLog.mainSection.uniqueIdentifier,
+            duration: Int(xcactivityLog.mainSection.timeStoppedRecording * 1000) -
+                Int(xcactivityLog.mainSection.timeStartedRecording * 1000),
+            isCI: ciChecker.isCI(),
+            modelIdentifier: machineEnvironment.modelIdentifier(),
+            macOSVersion: machineEnvironment.macOSVersion,
+            scheme: environment.schemeName,
+            xcodeVersion: try await xcodeBuildController.version()?.description
+        )
+        ServiceContext.current?.ui?.success(
+            .alert(
+                "Uploaded a build to the server."
+            )
+        )
+    }
+
+    private func latestXCActivityLog(
+        logManifestPlistPath: AbsolutePath,
+        buildLogsPath: AbsolutePath,
+        projectPath: AbsolutePath,
+        referenceDate: Date
+    ) async throws -> XCActivityLog {
+        let plist: LogStoreManifest = try await fileSystem
+            .readPlistFile(at: logManifestPlistPath)
+
+        guard let latestLog = plist.logs.values.sorted(by: { $0.timeStoppedRecording > $1.timeStoppedRecording }).first,
+              environment
+              .workspacePath == nil ||
+              (referenceDate.timeIntervalSinceReferenceDate - 10 ..< referenceDate.timeIntervalSinceReferenceDate + 10) ~=
```

Comment:
We don't want to upload an old xcactivitylog. We're assuming we get to this point in at least ten seconds after the `.xcactivitylog` is generated. In the worst case, the upload action is idempotent, so it's not the end of the world when we reupload an existing log.

I found that when running the process in a background, the `.xcactivitylog` is present. Alternatively, when there is no recent activity log present, we can watch the log store manifest for changes using a tool like [this](https://github.com/jarrodtaylor/fs-diff-stream/tree/main) one. However, I'd only do it if we find that to be necessary.